### PR TITLE
feat: add header

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Added header component, refactored code

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [x] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Material UI components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change

image

image

https://user-images.githubusercontent.com/117748123/203555183-bdd484b3-e22a-40d2-9229-ba68c69ded61.mp4

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { UserPanel } from './UserPanel';
+
+export const Header = () => {
+  const linkStyles = {
+    display: 'flex',
+    alignItems: 'center'
+  };
+
+  return (
+    <>
```
Comment (ahodkov) [inline src/components/Header/Header.jsx]: Думаю, что это излишне. <Box> это же единственный child здесь

File: src/components/Header/UserInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      sx={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}
+    >
+      <Avatar
+        sx={{
+          mr: '10px',
+          width: '40px',
+          height: '40px',
+          bgcolor: yellow[700]
+        }}
+        alt="Ivan Ivanov"
```
Comment (ahodkov) [inline src/components/Header/UserInfo.jsx]: alt="Profile Image"

File: src/components/Header/UserInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Avatar, Box, Typography } from '@mui/material';
+import { yellow } from '@mui/material/colors';
+
+import userImage from '../../assets/img/cat.jpg';
+
+export const UserInfo = () => {
```
Comment (ahodkov) [inline src/components/Header/UserInfo.jsx:7]: Наверное лучше добавить сразу параметры и просто его передавать:
`export const UserInfo = ({imageSrc, fullName, title}) => {}`

Comment (ahodkov) [reply src/components/Header/UserInfo.jsx:7]: и добавить PropTypes

File: src/components/RouterLink/RouterLink.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Link as LinkReact } from 'react-router-dom';
+import { Link } from '@mui/material';
+
+export const RouterLink = ({ children, linkPath, stylesObj }) => {
```
Comment (ahodkov) [inline src/components/RouterLink/RouterLink.jsx:6]: PropTypes

File: src/components/ContentContainer/ContentContainer.jsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ export const ContentContainer = ({ children, isLoginPage = false }) => {
         flexDirection: 'column',
```
Comment (ahodkov) [inline src/components/ContentContainer/ContentContainer.jsx:10]: ContanetContainer тоже добавить PropTypes

File: src/components/ContentWrapper/ContentWrapper.jsx [hunk 6]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 import { Box } from '@mui/material';
 
-export const ContentWrapper = ({ children, type = 'div' }) => {
+export const ContentWrapper = ({ children, type = 'div', stylesObj = {} }) => {
```
Comment (ahodkov) [inline src/components/ContentWrapper/ContentWrapper.jsx:5]: PropTypes

File: src/components/Header/UserPanel.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import NotificationsIcon from '@mui/icons-material/Notifications';
+import SettingsIcon from '@mui/icons-material/Settings';
```
Comment (ahodkov) [inline src/components/Header/UserPanel.jsx]: если в icons-material есть re-импорт в index.js, то есть смысл сделать импорт в одну строчку.
То есть:
`import {NotificationsIcon, SettingsIcon} from '@mui/icons-material'`

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: src/components/Header/UserPanel.jsx (hunk 7 above)
Comment (ahodkov) [reply src/components/Header/UserPanel.jsx]: https://mui.com/material-ui/icons/#usage option 2

File: src/components/Header/UserInfo.jsx (hunk 3 above)
Comment (ahodkov) [reply src/components/Header/UserInfo.jsx:7]: https://reactjs.org/docs/typechecking-with-proptypes.html

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (LBokan) [reply src/components/Header/Header.jsx]: Убрала <> и </>
image

File: src/components/Header/UserInfo.jsx (hunk 2 above)
Comment (LBokan) [reply src/components/Header/UserInfo.jsx]: image

File: src/components/Header/UserInfo.jsx (hunk 3 above)
Comment (LBokan) [reply src/components/Header/UserInfo.jsx:7]: image

File: src/components/RouterLink/RouterLink.jsx (hunk 4 above)
Comment (LBokan) [reply src/components/RouterLink/RouterLink.jsx:6]: image

File: src/components/ContentContainer/ContentContainer.jsx (hunk 5 above)
Comment (LBokan) [reply src/components/ContentContainer/ContentContainer.jsx:10]: image

File: src/components/ContentWrapper/ContentWrapper.jsx (hunk 6 above)
Comment (LBokan) [reply src/components/ContentWrapper/ContentWrapper.jsx:5]: image

File: src/components/Header/UserPanel.jsx (hunk 7 above)
Comment (LBokan) [reply src/components/Header/UserPanel.jsx]: image
image
image

File: package.json [hunk 8]
```diff
@@ -2,7 +2,7 @@
   "name": "SmartHome",
   "main": "index.js",
   "scripts": {
-    "start": "cross-env NODE_ENV=production webpack-dev-server",
+    "start": "cross-env NODE_ENV=development webpack-dev-server",
```
Comment (ahodkov) [inline package.json:5]: Можно добавить
"start:prod": "cross-end NODE_ENV=production webpack-dev-server"

**Review by ahodkov — APPROVED** [review]

File: package.json (hunk 8 above)
Comment (LBokan) [reply package.json:5]: Ок, добавлю в след. PR

**Pull request merged**